src/realm/obj.cpp

```diff
@@ -1936,6 +1936,84 @@ bool Obj::remove_backlink(ColKey col_key, ObjLink old_link, CascadeState& state)
 
 void Obj::assign(const Obj& other)
 {
+    copy(other);
+    auto copy_links = [this, &other](ColKey col) {
+        auto t = m_table->get_opposite_table(col);
+        auto c = m_table->get_opposite_column(col);
+        auto backlinks = other.get_all_backlinks(col);
+        for (auto bl : backlinks) {
+            auto linking_obj = t->get_object(bl);
+            if (c.get_type() == col_type_LinkList) {
+                auto linking_obj_list = linking_obj.get_linklist(c);
+                auto n = linking_obj_list.find_first(other.get_key());
+                REALM_ASSERT(n != realm::npos);
+                linking_obj_list.set(n, get_key());
+            }
+            else if (c.get_attrs().test(col_attr_Set)) {
+                REALM_UNREACHABLE();
```

Comment:
Since `Obj::assign` is meant to be used for normal object types, this might be a set of _non-embedded_ objects which we should add support for.

Reply:
this should now be resolved.